drivers/tee/tee_shm.c

```diff
@@ -221,8 +330,12 @@ void tee_shm_free(struct tee_shm *shm)
 	 *
 	 * In the case of driver private memory we call tee_shm_release
 	 * directly instead as it doesn't have a reference counter.
+	 *
+	 * In case the buffer was registered (not mappable), we call
+	 * tee_shm_release directly as the initial dmabuf reference must
+	 * be released before actual shm object to released.
```

Comment:
By calling `tee_shm_release()` the actual shm object is released before the dmabuf reference is released.